tests/simulate/test_simulate.py

```diff
@@ -43,6 +43,9 @@ def derivative(self, key, var, x):
             d = d.item()
         return d
 
+    def _validate_coordinate(self, x):
+        pass
```

Comment:
If you are implementing this as empty, it needs to return `x` based on the signature you're using
```suggestion
        return x
```

Reply:
Given the change above to not have it pass the value back, I think this is now fine.